src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultApplicationModelProvider.cs

```diff
@@ -433,25 +433,31 @@ protected virtual ParameterModel CreateParameterModel(ParameterInfo parameterInf
                 throw new ArgumentNullException(nameof(parameterInfo));
             }
 
-            // CoreCLR returns IEnumerable<Attribute> from GetCustomAttributes - the OfType<object>
-            // is needed to so that the result of ToArray() is object
             var attributes = parameterInfo.GetCustomAttributes(inherit: true);
-            var parameterModel = new ParameterModel(parameterInfo, attributes);
+            var parameterModel = new ParameterModel(parameterInfo, attributes)
+            {
+                ParameterName = parameterInfo.Name
+            };
 
-            var bindingInfo = BindingInfo.GetBindingInfo(attributes);
-            if (bindingInfo != null)
+            BindingInfo bindingInfo;
+            if (_modelMetadataProvider is ModelMetadataProvider modelMetadataProviderBase)
+            {
+                var modelMetadata = modelMetadataProviderBase.GetMetadataForParameter(parameterInfo);
```

Comment:
Never make this call (or down-cast to `ModelMetadataProvider`) without checking `MvcOptions.AllowValidatingTopLevelNodes`.